V1.91 Compiles on Mac But with Errors

So I managed to compile V1.91 on a Mac, but I get some very weird results...
FYI: I used cmake for this

1. The outputted file in not a .APP like the v1.5 version (but I think you say that in the instructions).
2. libpng warning: iCCP: known incorrect sRGB profile
3. You have to MOGRIFY the PNGs in the resources folder, or...
4. ...you get libpng errors when running the Unix executable that is output by the make process
5. The outputted file is a fifth of the size of its predecessor (3.9mb).
6. Warnings are produced that OPERATOR and CREATED are deprecated
7. For qmake, `make lcopy` doesn't work `make: *** No rule to make target 'lcopy'.  Stop.`
8. You seem to have to `qmake` in the `SRC` folder then `make lcopy` works, but i still have to run `make` to get an output
9. During the qmake process `datautils.c:26:10: fatal error: 'libxml/xmlmemory.h' file not found`
10. During the qmake process `clang: error: linker command failed with exit code 1 (use -v to see invocation)`

These errors recur a lot;
`
vitamtp.c:603:20: warning: unused variable 'data' [-Wunused-variable]
    unsigned char *data = NULL;
                                ^
vitamtp.c:63:26: warning: inline function 'VitaMTP_Get_PTP_Params' is not defined [-Wundefined-inline]
extern inline PTPParams *VitaMTP_Get_PTP_Params(vita_device_t *device);
                                              ^
vitamtp.c:74:25: note: used here
    PTPParams *params = VitaMTP_Get_PTP_Params(device);
                                             ^
`

`
psvlib/pkg/pkg_sys.c:211:13: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
            _error_func(NULL, "ERROR: cannot create '%s' folder\n", path);
            ^
psvlib/pkg/pkg_sys.c:221:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: cannot open '%s' file\n", fname);
        ^
psvlib/pkg/pkg_sys.c:227:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: cannot get size of '%s' file\n", fname);
        ^
psvlib/pkg/pkg_sys.c:239:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: cannot create '%s' file\n", fname);
        ^
psvlib/pkg/pkg_sys.c:249:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: failed to close file\n");
        ^
psvlib/pkg/pkg_sys.c:258:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: failed to read %u bytes from file\n", size);
        ^
psvlib/pkg/pkg_sys.c:267:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: failed to read %u bytes from file\n", size);
        ^
psvlib/pkg/pkg_sys.c:303:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: internal error, wrong sys_realloc usage\n");
        ^
psvlib/pkg/pkg_sys.c:308:9: warning: implicit declaration of function '_error_func' is invalid in C99 [-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: out of memory\n");
        ^
psvlib/pkg/pkg_sys.c:345:9: warning: implicit declaration of function '_output_func' is invalid in C99 [-Wimplicit-function-declaration]
        _output_func(arg, "[*] unpacking... %u%%\r", now);
        ^
`

`
psvlib/pkg/pkg_zrif.c:53:22: warning: assigning to 'Bytef *' (aka 'unsigned char *') from 'const Bytef *' (aka 'const unsigned char *') discards qualifiers
      [-Wincompatible-pointer-types-discards-qualifiers]
    d_stream.next_in = (const Bytef*)in;
                                   ^ ~~~~~~~~~~~~
`

@juanabeja 

do you mind sharing in how you avercame this error during 'make'

```datautils.c:26:10: fatal error: 'libxml/xmlmemory.h' file not found
#include <libxml/xmlmemory.h>
         ^~~~~~~~~~~~~~~~~~~~
```

> @juanabeja
> 
> do you mind sharing in how you avercame this error during 'make'
> 
> ```
> datautils.c:26:10: fatal error: 'libxml/xmlmemory.h' file not found
> #include <libxml/xmlmemory.h>
>          ^~~~~~~~~~~~~~~~~~~~
> ```

Never got this error, so I'm not sure. `qmake` and `make` were all it took for me once I installed pkg-config.

Hi there,
some findings I have made, which could be added to readme (not proposing a PR because I couldn't compile it until the end):

1. cmake needs to be installed manually - `brew install cmake`
2. cmake tries to find qt5, not qt(4), so `brew install qt5`
3. it is not symlinked, so run cmake with `cmake -DCMAKE_PREFIX_PATH=/usr/local/opt/qt@5 .`

Then the configuration part is apparently done:

```
-- Configuring done
CMake Warning (dev) in src/CMakeLists.txt:
  Policy CMP0100 is not set: Let AUTOMOC and AUTOUIC process .hh files.  Run
  "cmake --help-policy CMP0100" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  For compatibility, CMake is excluding the header file(s):

    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/downloader.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/finalhe.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/package.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/sforeader.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/vita.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/worker.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/worker.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/sforeader.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/package.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/finalhe.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/vita.hh"
    "/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/src/downloader.hh"

  from processing by AUTOMOC.  If any of the files should be processed, set
  CMP0100 to NEW.  If any of the files should not be processed, explicitly
  exclude them by setting the source file property SKIP_AUTOMOC:

    set_property(SOURCE file.hh PROPERTY SKIP_AUTOMOC ON)

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/arlindo.pereira/Downloads/Vita/finalhe-1.92
```

Then when I try to `make` it:

```
Scanning dependencies of target miniz
[  1%] Building C object deps/miniz/CMakeFiles/miniz.dir/miniz.c.o
[  3%] Building C object deps/miniz/CMakeFiles/miniz.dir/miniz_zip.c.o
[  4%] Building C object deps/miniz/CMakeFiles/miniz.dir/miniz_tinfl.c.o
[  6%] Building C object deps/miniz/CMakeFiles/miniz.dir/miniz_tdef.c.o
[  7%] Linking C static library libminiz.a
[  7%] Built target miniz
Scanning dependencies of target scrypto
[  9%] Building C object deps/scrypto/CMakeFiles/scrypto.dir/aes.c.o
[ 11%] Building C object deps/scrypto/CMakeFiles/scrypto.dir/aes_x86.c.o
[ 12%] Building C object deps/scrypto/CMakeFiles/scrypto.dir/base64.c.o
[ 14%] Building C object deps/scrypto/CMakeFiles/scrypto.dir/sc_crc32.c.o
[ 15%] Building C object deps/scrypto/CMakeFiles/scrypto.dir/sc_crc32_x86.c.o
[ 17%] Building C object deps/scrypto/CMakeFiles/scrypto.dir/sha256.c.o
[ 19%] Linking C static library libscrypto.a
[ 19%] Built target scrypto
Scanning dependencies of target vitamtp
[ 20%] Building C object deps/vitamtp/CMakeFiles/vitamtp.dir/datautils.c.o
[ 22%] Building C object deps/vitamtp/CMakeFiles/vitamtp.dir/device.c.o
[ 23%] Building C object deps/vitamtp/CMakeFiles/vitamtp.dir/ptp.c.o
[ 25%] Building C object deps/vitamtp/CMakeFiles/vitamtp.dir/usb.c.o
[ 26%] Building C object deps/vitamtp/CMakeFiles/vitamtp.dir/vitamtp.c.o
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/vitamtp/vitamtp.c:603:20: warning: 
      unused variable 'data' [-Wunused-variable]
    unsigned char *data = NULL;
                   ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/vitamtp/vitamtp.c:63:26: warning: 
      inline function 'VitaMTP_Get_PTP_Params' is not defined
      [-Wundefined-inline]
extern inline PTPParams *VitaMTP_Get_PTP_Params(vita_device_t *device);
                         ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/vitamtp/vitamtp.c:74:25: note: 
      used here
    PTPParams *params = VitaMTP_Get_PTP_Params(device);
                        ^
2 warnings generated.
[ 28%] Building C object deps/vitamtp/CMakeFiles/vitamtp.dir/wireless.c.o
[ 30%] Linking C static library libvitamtp.a
[ 30%] Built target vitamtp
Scanning dependencies of target vitapkg
[ 31%] Building C object deps/psvlib/pkg/CMakeFiles/vitapkg.dir/pkg.c.o
[ 33%] Building C object deps/psvlib/pkg/CMakeFiles/vitapkg.dir/pkg_out.c.o
[ 34%] Building C object deps/psvlib/pkg/CMakeFiles/vitapkg.dir/pkg_sys.c.o
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:211:13: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
            _error_func(NULL, "ERROR: cannot create '%s' folder\n", path);
            ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:221:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: cannot open '%s' file\n", fname);
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:227:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: cannot get size of '%s' file\n", fname);
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:239:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: cannot create '%s' file\n", fname);
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:249:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: failed to close file\n");
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:258:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: failed to read %u bytes from file\n", size);
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:267:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: failed to read %u bytes from file\n", size);
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:303:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: internal error, wrong sys_realloc usage\n");
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:308:9: error: 
      implicit declaration of function '_error_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _error_func(NULL, "ERROR: out of memory\n");
        ^
/Users/arlindo.pereira/Downloads/Vita/finalhe-1.92/deps/psvlib/pkg/pkg_sys.c:345:9: error: 
      implicit declaration of function '_output_func' is invalid in C99
      [-Werror,-Wimplicit-function-declaration]
        _output_func(arg, "[*] unpacking... %u%%\r", now);
        ^
10 errors generated.
make[2]: *** [deps/psvlib/pkg/CMakeFiles/vitapkg.dir/pkg_sys.c.o] Error 1
make[1]: *** [deps/psvlib/pkg/CMakeFiles/vitapkg.dir/all] Error 2
make: *** [all] Error 2
```

Any idea why that happens? Thanks in advance.

Trying to compile on Big Sur, getting
`Undefined symbols for architecture x86_64:
  "Downloader::finishedGet(void*)", referenced from:
      Downloader::downloadFinished(QNetworkReply*) in downloader.cc.o`
`...`
`ld: symbol(s) not found for architecture x86_64`
`clang: error: linker command failed with exit code 1 (use -v to see invocation)`
`make[2]: *** [src/FinalHE] Error 1`
`make[1]: *** [src/CMakeFiles/FinalHE.dir/all] Error 2`


Anyone knows how to resolve? Note: also had to add 
`set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=implicit-function-declaration")
`
to  CMakeLists.txt because I was getting
`error: 
      implicit declaration of function '_error_func' is invalid in C99`
as an error, not a warning, so not sure if this is related to my compiler configuration...

--- a/README.md
+++ b/README.md
@@ -34,8 +34,8 @@ Download a pre-built executable binary below and follow instructions. Supported 
 
 ### Prerequisites 
 
-1. macOS: install [brew](https://brew.sh), install libusb, pkg-config & qt through brew (`brew install libusb pkg-config qt`)
-2. Linux:
+1. macOS: install [brew](https://brew.sh), build tools (`xcode-select --install`) then install dyld, libusb, pkg-config & qt5 through brew (`brew install dyld-headers libusb pkg-config qt5`)
+1. Linux:
    - Debian/Ubuntu: install build-essential, libxml2-dev, libusb-dev, libusb-1.0-0-dev, zlib-dev or zlib1g-dev, qtbase5-dev, qttools5, cmake(if use cmake to build)
    - Fedora/CentOS: group install "Development Tools", install libxml2-devel, libusb-devel, zlib-devel, qt5-qtbase-devel, qt5-qtbase, cmake3(if use cmake to build)
    - openSUSE: install cmake >= 11.0, gcc-c++, zlib-devel, libxml2-devel, libQt5Widgets-devel, libQt5Network-devel, libqt5-linguist-devel, libusb-compat-devel
@@ -46,7 +46,7 @@ Download a pre-built executable binary below and follow instructions. Supported 
 You can choose either `qmake` or `cmake` to build
 
 - cmake: run `cmake` to generate Makefile for compiling
-  - macOS: it cannot produce app bundle, and you need to specify `CMAKE_PREFIX_PATH` if Qt is not installed in default location: `cmake -DCMAKE_PREFIX_PATH=<Path of Qt Root> <Path of Project Root>`
+  - macOS: it cannot produce app bundle, and you need to specify `CMAKE_PREFIX_PATH` if Qt is not installed in default location: `cmake -DCMAKE_PREFIX_PATH=<Path of Qt Root> <Path of Project Root>`. If you have installed qt5 with brew, and are on project root directory, you can run with `cmake -DCMAKE_PREFIX_PATH=/usr/local/opt/qt@5 .`. Then run `make` and you can open it by double clicking at `src/FinalHE`.
 - qmake: run `qmake` to generate Makefile for compiling, run `make lcopy` in `src` folder to compile translations and copy them to binary folder
 
 ## Contribute translations

